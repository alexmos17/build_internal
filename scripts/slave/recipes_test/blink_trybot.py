# Copyright 2013 The Chromium Authors. All rights reserved.
# Use of this source code is governed by a BSD-style license that can be
# found in the LICENSE file.

"""Test inputs for recipes/blink_trybot.py"""

SUCCESS_DATA = {}

FAIL_DATA = {
  'webkit_tests (with patch)': (1, {'crazy': ['data', 'format']}),
  'webkit_tests (without patch)': (1, {'crazy': ['data', 'format']})
}

for result in ['success', 'fail']:
  for build_config in ['Release', 'Debug']:
    def closure(result_, build_config_):
      return lambda api: {
        'build_properties': api.build_properties_tryserver(
          build_config=build_config_,
          config_name='blink',
          root='src/third_party/WebKit',
        ),
        'test_data': globals()['%s_DATA' % result_.upper()]
      }
    ret = closure(result, build_config)
    globals()['%s_%s_test' % (result, build_config.lower())] = ret

