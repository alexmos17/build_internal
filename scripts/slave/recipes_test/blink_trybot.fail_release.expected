[
  {
   "cmd": ["[DEPOT_TOOLS_ROOT]/gclient", "config", "--spec", "deps_os = {}\nhooks = []\ntarget_os_only = False\nsolutions = [{'managed': True, 'custom_vars': {'webkit_trunk': 'svn://svn-mirror.golo.chromium.org/blink/trunk', 'googlecode_url': 'svn://svn-mirror.golo.chromium.org/%s', 'nacl_trunk': 'svn://svn-mirror.golo.chromium.org/native_client/trunk', 'sourceforge_url': 'svn://svn-mirror.golo.chromium.org/%(repo)s'}, 'url': 'svn://svn-mirror.golo.chromium.org/chrome/trunk/src', 'custom_deps': {'src/third_party/WebKit/LayoutTests': None, 'src/webkit/data/layout_tests/LayoutTests': None, 'src/third_party/WebKit': 'svn://svn-mirror.golo.chromium.org/blink/trunk'}, 'deps_file': 'DEPS', 'safesync_url': None, 'name': 'src'}]\ntarget_os = []\n"],
   "name": "gclient setup",
   "seed_steps": ["gclient setup", "gclient revert", "gclient sync", "apply_issue", "gclient runhooks", "compile", "webkit_unit_tests", "webkit_lint"]
  },
  {
   "cmd": ["python", "[BUILD_ROOT]/scripts/slave/gclient_safe_revert.py", ".", "[DEPOT_TOOLS_ROOT]/gclient"],
   "name": "gclient revert"
  },
  {
   "cmd": ["[DEPOT_TOOLS_ROOT]/gclient", "sync", "--nohooks"],
   "name": "gclient sync"
  },
  {
   "cmd": ["[DEPOT_TOOLS_ROOT]/apply_issue", "-r", "[SLAVE_BUILD_ROOT]/src/third_party/WebKit", "-i", 12853011, "-p", 1, "-s", "https://chromiumcodereview.appspot.com", "-e", "commit-bot@chromium.org"],
   "name": "apply_issue"
  },
  {
   "cmd": ["[DEPOT_TOOLS_ROOT]/gclient", "runhooks"],
   "env": {"GYP_DEFINES": "fastbuild=1 component=static_library", "GYP_GENERATORS": "ninja", "GYP_GENERATOR_FLAGS": "", "GYP_MSVS_VERSION": ""},
   "name": "gclient runhooks"
  },
  {
   "cmd": ["python", "[BUILD_ROOT]/scripts/slave/compile.py", "--target", "Release", "--build-dir", "[SLAVE_BUILD_ROOT]/src/out", "--build-tool", "ninja", "--compiler", "goma", "--", "all_webkit", "content_shell"],
   "name": "compile"
  },
  {
   "cmd": ["python", "[BUILD_ROOT]/scripts/slave/runtest.py", "--target", "Release", "--build-dir", "[SLAVE_BUILD_ROOT]/src/out", "--no-xvfb", "--factory-properties", "{\"workdir\": \"/path/to/workdir/TestSlavename\", \"repository\": \"\", \"buildername\": \"TestBuilder\", \"recipe\": \"blink_trybot\", \"mastername\": \"chromium.testing.master\", \"config_name\": \"blink\", \"requester\": \"commit-bot@chromium.org\", \"build_config\": \"Release\", \"buildnumber\": 571, \"rietveld\": \"https://chromiumcodereview.appspot.com\", \"root\": \"src/third_party/WebKit\", \"project\": \"chrome\", \"slavename\": \"TestSlavename\", \"blamelist\": [\"cool_dev1337@chromium.org\", \"hax@chromium.org\"], \"branch\": \"\", \"patchset\": 1, \"issue\": 12853011, \"revision\": \"HEAD\"}", "--build-properties", "{\"workdir\": \"/path/to/workdir/TestSlavename\", \"repository\": \"\", \"buildername\": \"TestBuilder\", \"recipe\": \"blink_trybot\", \"mastername\": \"chromium.testing.master\", \"config_name\": \"blink\", \"requester\": \"commit-bot@chromium.org\", \"build_config\": \"Release\", \"buildnumber\": 571, \"rietveld\": \"https://chromiumcodereview.appspot.com\", \"root\": \"src/third_party/WebKit\", \"project\": \"chrome\", \"slavename\": \"TestSlavename\", \"blamelist\": [\"cool_dev1337@chromium.org\", \"hax@chromium.org\"], \"branch\": \"\", \"patchset\": 1, \"issue\": 12853011, \"revision\": \"HEAD\"}", "webkit_unit_tests"],
   "name": "webkit_unit_tests"
  },
  {
   "cmd": ["python", "[BUILD_ROOT]/scripts/slave/chromium/lint_test_files_wrapper.py", "--build-dir", "[SLAVE_BUILD_ROOT]/src/out", "--target", "Release"],
   "name": "webkit_lint"
  },
  {
   "can_fail_build": false,
   "cmd": ["python", "[BUILD_ROOT]/scripts/slave/chromium/layout_test_wrapper.py", "--target", "Release", "-o", "[SLAVE_BUILD_ROOT]/layout-test-results", "--build-dir", "[SLAVE_BUILD_ROOT]/src/out", "--output-json", "/path/to/tmp/json"],
   "name": "webkit_tests (with patch)"
  },
  {
   "cmd": ["python", "[BUILD_ROOT]/scripts/slave/gclient_safe_revert.py", ".", "[DEPOT_TOOLS_ROOT]/gclient"],
   "name": "gclient revert (2)",
   "seed_steps": ["gclient revert (2)", "gclient runhooks (2)", "compile (2)", "webkit_tests (without patch)"]
  },
  {
   "cmd": ["[DEPOT_TOOLS_ROOT]/gclient", "runhooks"],
   "env": {"GYP_DEFINES": "fastbuild=1 component=static_library", "GYP_GENERATORS": "ninja", "GYP_GENERATOR_FLAGS": "", "GYP_MSVS_VERSION": ""},
   "name": "gclient runhooks (2)"
  },
  {
   "cmd": ["python", "[BUILD_ROOT]/scripts/slave/compile.py", "--target", "Release", "--build-dir", "[SLAVE_BUILD_ROOT]/src/out", "--build-tool", "ninja", "--compiler", "goma", "--", "all_webkit", "content_shell"],
   "name": "compile (2)"
  },
  {
   "can_fail_build": false,
   "cmd": ["python", "[BUILD_ROOT]/scripts/slave/chromium/layout_test_wrapper.py", "--target", "Release", "-o", "[SLAVE_BUILD_ROOT]/layout-test-results", "--build-dir", "[SLAVE_BUILD_ROOT]/src/out", "--output-json", "/path/to/tmp/json"],
   "name": "webkit_tests (without patch)"
  },
  {
   "cmd": ["python", "[SLAVE_BUILD_ROOT]/src/third_party/WebKit/Tools/Scripts/print-json-test-results", "--ignored-failures-path", "{\"crazy\": [\"data\", \"format\"]}", "{\"crazy\": [\"data\", \"format\"]}"],
   "name": "webkit_tests"
  }
]